dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -93,4 +96,14 @@ def report_abuse_form
       age: (current_user.age unless current_user.nil?),
     }
   end
+
+  private
+
+  def check_if_featured(project_id)
+    _, channel_id = storage_decrypt_channel_id(project_id)
+    return render_404 unless channel_id
```

Comment:
I don't think this will work as intended.  `storage_decrypt_channel_id` will raise if anything is wrong, rather than return a falsy value - so you'll need to `rescue` to handle this case.

https://github.com/code-dot-org/code-dot-org/blob/05908d48ec4b36064a60d5a704086ac27257bc31/shared/middleware/helpers/storage_id.rb#L39-L47

That said, this 404 on invalid channel id is a new behavior, and I wonder if we'd prefer to just answer `false`, no it's not featured, and let the Zendesk ticket be filed in this case.  I think that would match the existing behavior on a bad channel ID.  Maybe we should write an integration test for that.

Reply:
And since it appears to only raise `ArgumentError`, rescue that type specifically.